addons/im_livechat/static/src/embed/core/livechat_service.js

```diff
@@ -145,24 +156,45 @@ export class LivechatService {
         let session = JSON.parse(this.cookie.current[this.SESSION_COOKIE] ?? false);
         if (session?.uuid && this.state === SESSION_STATE.NONE) {
             // Channel is already created on the server.
-            session.messages = await this.rpc("/im_livechat/chat_history", {
-                uuid: session.uuid,
-            });
-            session.messages.reverse();
-            this.busService.addChannel(session.uuid);
+            this.state = SESSION_STATE.PERSISTED;
+            const [messages] = await Promise.all([
+                this.rpc("/im_livechat/chat_history", {
+                    uuid: session.uuid,
+                }),
+                await this.initializePersistedSession(),
+            ]);
+            session.messages = messages.reverse();
```

Comment:
Out-of-scope of PR but something to improve in follow-up PR:
This function sucks by its naming and what it does: a `getX` function should not have so many (and generally even no) side-effects.

By their function names, `getLivechatThread` should not `getSession` and have all these side-effects.
At a glance, `getLivechatThread` and `getSession` (and `initializePersistedSession`) and `_createSession` should all be merged in a single function `start/make/insertSession`.